packages/@uppy/companion-client/src/RequestClient.js

```diff
@@ -16,35 +16,31 @@ async function handleJSONResponse (res) {
     throw new AuthError()
   }
 
-  const jsonPromise = res.json()
-
-  if (res.status < 200 || res.status > 300) {
-    let errMsg = `Failed request with status: ${res.status}. ${res.statusText}`
-    try {
-      const errData = await jsonPromise
-      errMsg = errData.message ? `${errMsg} message: ${errData.message}` : errMsg
-      errMsg = errData.requestId ? `${errMsg} request-Id: ${errData.requestId}` : errMsg
-    } finally {
-      // eslint-disable-next-line no-unsafe-finally
-      throw new Error(errMsg)
-    }
+  // todo why is 300 considered success?
+  // shouldn't we use res.ok?
+  if (res.status >= 200 && res.status <= 300) {
+    return res.json()
   }
-  return jsonPromise
+
+  let errMsg = `Failed request with status: ${res.status}. ${res.statusText}`
+  const errData = await res.json()
+  errMsg = errData.message ? `${errMsg} message: ${errData.message}` : errMsg
+  errMsg = errData.requestId ? `${errMsg} request-Id: ${errData.requestId}` : errMsg
+  throw new Error(errMsg)
 }
 
+// todo pull out into core instead?
```

Comment:
core has no knowledge of whether you use Companion or not so I don't think it belongs there

Reply:
How about `this.getPluginState`/`this.setPluginState`?

Reply:
That's a good idea, but we would have to send `this.getPluginState` and `this.setPluginState` into every RequestClient instance. Not sure if that's the cleanest way to do it.